contracts/OtokenFactory.sol

```diff
@@ -0,0 +1,185 @@
+pragma solidity =0.6.10;
+
+import {OtokenSpawner} from "./OtokenSpawner.sol";
+import {AddressBookInterface} from "./interfaces/AddressBookInterface.sol";
+import {OtokenInterface} from "./interfaces/OtokenInterface.sol";
+import {WhitelistInterface} from "./interfaces/WhitelistInterface.sol";
+
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ * @title A factory for opyn tokens
+ * @author Opyn
+ * @notice Create new otokens and keep track of all created tokens.
+ * @dev Calculate contract address before each creation with CREATE2
+ * and deploy eip-1167 minimal proxies for otoken logic contract.
+ */
+contract OtokenFactory is OtokenSpawner {
+    /// @notice The Opyn AddressBook contract that records addresses of whitelist module and otoken impl address. */
+    address public addressBook;
+
+    /// @dev An array of all created otokens */
+    address[] private _otokens;
+
+    /// @dev A mapping from parameters hash to its deployed address
+    mapping(bytes32 => address) private _idToAddress;
```

Comment:
same as above

Reply:
You're right we don't need `_` for private variables.